src/credentials.rs

```diff
@@ -32,54 +38,78 @@ pub struct Credentials {
 }
 
 impl Credentials {
-    fn location() -> eyre::Result<PathBuf> {
-        let home = home::home_dir().wrap_err("Failed to locate home directory")?;
+    /// Get buffrs config directory.
+    fn buffrs_home() -> Result<PathBuf, VarError> {
+        var("BUFFRS_HOME").map(PathBuf::from)
+    }
 
-        let home = env::var("BUFFRS_HOME").map(PathBuf::from).unwrap_or(home);
+    /// XDG path: uses whichever config directory is appropriate for the platform.
+    ///
+    /// For example, on Linux this might be `~/.config/buffrs/credentials.toml`.
+    fn xdg_path() -> PathBuf {
+        Self::buffrs_home()
+            .unwrap_or_else(|_| dirs::config_dir().expect("get config dir").join(BUFFRS_DIR))
+            .join(CREDENTIALS_FILE)
+    }
 
-        Ok(home.join(BUFFRS_HOME).join(CREDENTIALS_FILE))
+    /// Legacy path: hard-coded to `~/.buffrs/credentials.toml`.
+    fn legacy_path() -> PathBuf {
+        Self::buffrs_home()
+            .unwrap_or_else(|_| dirs::home_dir().expect("get home dir").join(BUFFRS_HOME))
+            .join(CREDENTIALS_FILE)
+    }
+
+    /// Possible locations for the credentials file
+    fn possible_locations() -> Vec<PathBuf> {
+        vec![Self::xdg_path(), Self::legacy_path()]
     }
 
     /// Checks if the credentials exists
     pub async fn exists() -> eyre::Result<bool> {
-        fs::try_exists(Self::location()?)
-            .await
-            .wrap_err("Failed to detect credentials")
+        for location in &Self::possible_locations() {
+            if fs::try_exists(&location).await? {
```

Comment:
Do we really want to terminate on the first error? Imagine we fail to access the system-wide credentials because we lack permission, shouldn't we keep going and try user locations too?

BTW `fs::try_exists` [is only in nightly](https://doc.rust-lang.org/stable/std/fs/fn.try_exists.html) so I'm surprised this compiles in CI (probably something worth looking into, will file an issue for it).

Reply:
Womp, you're right! This is why I don't like tokio shadowing std stuff... 😅  Ignore that part (and I'll close the issue).